src/metabase/query_processor/middleware/visualization_settings.clj

```diff
@@ -17,12 +17,30 @@
   "For each field, fetch its settings from the QP store, convert the settings into the normalized form
   for visualization settings, and then merge in the card-level column settings."
   [column-viz-settings field-ids]
-  (merge column-viz-settings
-         (into {} (for [field-id field-ids]
-                    (let [field-settings      (:settings (lib.metadata/field (qp.store/metadata-provider) field-id))
-                          norm-field-settings (normalize-field-settings field-id field-settings)
-                          col-settings        (get column-viz-settings {::mb.viz/field-id field-id})]
-                      [{::mb.viz/field-id field-id} (merge norm-field-settings col-settings)])))))
+  ;; Retrieve field-level settings
+  (let [field-id->settings      (reduce
+                                  (fn [m field-id]
+                                    (let [field-settings      (:settings (lib.metadata/field (qp.store/metadata-provider) field-id))
+                                          norm-field-settings (normalize-field-settings field-id field-settings)]
+                                      (cond-> m
+                                        (seq norm-field-settings)
+                                        (assoc field-id norm-field-settings))))
+                                  {}
+                                  field-ids)
+        ;; For each column viz setting, if there is a match on the field settings, merge it in,
+        ;; with the column viz settings being the default in the event of conflicts.
+        merged-settings         (reduce-kv
+                                  (fn [coll {field-id ::mb.viz/field-id :as k} column-viz-setting]
+                                    (assoc coll k (merge (get field-id->settings field-id {}) column-viz-setting)))
+                                  {}
+                                  column-viz-settings)
+        ;; The field-ids that are in the merged settings
+        viz-field-ids           (set (map ::mb.viz/field-id (keys merged-settings)))
+        ;; Keep any field settings that aren't in the merged settings and have settings
+        distinct-field-settings (update-keys
+                                  (remove (comp viz-field-ids first) field-id->settings)
+                                  (fn [k] {::mb.viz/field-id k}))]
+    (merge merged-settings distinct-field-settings)))
```

Comment:
Thanks for the comments in here, I think that's helpful! I know I'll forget the specifics in a few weeks anyway, so having them as a guide is nice :)